include/api/Metrics.h

```diff
@@ -45,23 +45,47 @@ class Metrics
 private:
   using self_type = Metrics;
 
+  class AtomicType
+  {
+    friend class Metrics;
+
+  public:
+    AtomicType()          = default;
+    virtual ~AtomicType() = default;
```

Comment:
Making this virtual doubles the size of instances of this type.  Why is it necessary?

Reply:
Same as above :/

Reply:
above what?

Reply:
It had to be virtual, otherwise I couldn't subclass it later as I did. And I had to sub class it, rather than aliases, to make the compiler detect bad usage.

Reply:
Maybe the right solution here is to have 

```
Metrics::Counter::lookup(stats_id);
Metrics::Gauge::lookup(stats_id);
```

?

Reply:
@ywkaras I added those static's, which allows us to eliminate the virtual. Let me know what you think, it's in the last commit that I lovingly made just for you.